src/main/scala/com/olegpy/meow/internal/HandleOptics.scala

```diff
@@ -0,0 +1,28 @@
+package com.olegpy.meow.internal
+
+import cats.mtl.ApplicativeHandle
+import com.olegpy.meow.optics.TPrism
+
+
+private[meow] object HandleOptics {
+  class Applicative[F[_], S, E](
+    parent: ApplicativeHandle[F, S],
+    prism: TPrism[S, E]
+  ) extends ApplicativeHandle[F, E] {
+    val applicative: cats.Applicative[F] = parent.applicative
+    def handle[A](fa: F[A])(f: E => A): F[A] =
+      parent.handle(fa) { case prism(e) => f(e) }
+
+    def handleWith[A](fa: F[A])(f: E => F[A]): F[A] =
+      parent.handleWith(fa) { case prism(e) => f(e) }
+
+    def attempt[A](fa: F[A]): F[Either[E, A]] =
+      functor.map(parent.attempt(fa)) {
+        case Right(a) => Right(a)
+        case Left(prism(e)) => Left(e)
```

Comment:
Same issue with pattern-match here and in `handle` - `map` takes a total function, but using `prism` as an extractor doesn't always match. You can, however, just extend `DefaultApplicativeHandle` to get the `attempt` and `handle` for free.

Reply:
Yep, I missed this. Fixed with `DefaultApplicativeHandle`.